### DISCUSSION TITLE
Getting responce status code issues (overriding OnResourceResponse)

### DISCUSSION THREAD

**Discussion by assopri** [cefsharp/CefSharp#3660] [General]
Hello! I override OnResourceResponse to get page responce status code. Faced 2 issues: 
	1. It works, however it fires many times, and I cannot figure out which is related to the main page load event (if such concept even exists). I supposed that could be identified by __ parameters, but I see they are all the same. Any suggestions here?
	2. When I use proxy, which (when loading through httpwebrequest library) return 408 code , I cannot get OnResourceResponse fired , thats why I cannot figure out what exactly error code I`ve got.

```c#
public class CustomResourceRequestHandler : CefSharp.Handler.ResourceRequestHandler
    {     
        protected override bool OnResourceResponse(IWebBrowser browserControl, IBrowser browser, IFrame frame, IRequest request, IResponse response)
        {
           
            // Console.WriteLine("HTTP Status Code: {0}", response.StatusCode);

            // allow the resource to load normally
            return false;
        }
    }
```

**Comment by amaitland** [comment]
> 1. It works, however it fires many times, and I cannot figure out which is related to the main page load event (if such concept even exists). I supposed that could be identified by __ parameters, but I see they are all the same. Any suggestions here?

OnResourceResponse should be called for every resource load, JavaScript, css, html, etc.

You should be able to tell the requests apart by looking at 
http://cefsharp.github.io/api/91.1.x/html/P_CefSharp_IRequest_ResourceType.htm
http://cefsharp.github.io/api/91.1.x/html/T_CefSharp_ResourceType.htm

> When I use proxy, which (when loading through httpwebrequest library) return 408 code , I cannot get OnResourceResponse fired , thats why I cannot figure out what exactly error code I`ve got.

I'm not sure what you mean by this.

**Reply by assopri** [reply]
Thank you! It works just fine!
